src/lib/csv-export/index.ts

```diff
@@ -0,0 +1,73 @@
+import {ModelPluginInterface} from '../../interfaces';
+import {ModelParams} from '../../types';
+
+import * as path from 'path';
+import * as fs from 'fs/promises';
+
+import {ERRORS} from '../../util/errors';
+import {buildErrorMessage} from '../../util/helpers';
+const {InputValidationError} = ERRORS;
+
+export class CsvExportModel implements ModelPluginInterface {
+  params: object | undefined;
+  outputPath: string | undefined;
+
+  errorBuilder = buildErrorMessage(CsvExportModel);
+
+  headers: string[] = [];
+
+  async configure(params: object | undefined): Promise<ModelPluginInterface> {
+    if (params === undefined) {
+      throw new InputValidationError(
+        this.errorBuilder({message: 'Configuration data is missing'})
+      );
+    }
+
+    this.params = params;
+
+    if ('output-path' in params) {
+      this.outputPath = params['output-path'] as string;
+    } else {
+      throw new InputValidationError(
+        this.errorBuilder({message: 'output-path required'})
+      );
+    }
+
+    if ('headers' in params) {
+      this.headers = params['headers'] as string[];
+    }
+
+    return this;
+  }
+
+  async execute(inputs: ModelParams[]): Promise<ModelParams[]> {
```

Comment:
not really sure about this, might be worth checking: we might want to have IO operations in this method (`fs.mkdir`, `fs.writeFile`) wrapped in a try-catch, so we can handle any IO issues and build errors with `this.errorBuilder` accordingly

Reply:
Indeed 🙂 thanks!